### ISSUE TITLE
Clean target should delete files under obj directory

### ISSUE THREAD

**Issue by AArnott** [dotnet/sdk#650]
```
git clone https://github.com/microsoft/vs-validation.git
cd vs-validation
git checkout acdf1788a
cd src\Microsoft.VisualStudio.Validation
msbuild /t:restore
msbuild
msbuild /t:clean

dir obj\Debug\net45
```

**Expected**

An empty directory

**Actual**

These files exist:

```
CoreCompileInputs.cache
Microsoft.VisualStudio.Validation.AssemblyInfo.cs
TemporaryGeneratedFile_036C0B5B-1481-4323-8D20-8F5ADCB23D92.cs
TemporaryGeneratedFile_5937a670-0e60-4077-877b-f7221da3dda1.cs
TemporaryGeneratedFile_E7A71F73-0F8D-4B9B-B56E-8E70B10BC5D3.cs
```

Of particular importance is the generated AssemblyInfo.cs file that remains, as that's a generated file and incremental build doesn't update it properly, so a clean build should definitely remove it.

**Comment by rainersigwald** [comment]
The first one is from MSBuild's common targets: https://github.com/Microsoft/msbuild/issues/1577.

**Comment by nguerrera** [comment]
@AArnott I'm not able to reproduce this either on d15prerel latest. The only file that remains is CoreCompileInputs.cache.

**Comment by nguerrera** [comment]
This may have been what fixed it: #276

Does it still repro for you on latest bits?

**Comment by AArnott** [comment]
It no longer repros for me with dotnet 1.0.0-rc4-004600.
Thanks.
